### PR TITLE
Merge Request A2-Sam

### PR DESCRIPTION
**Pull request by trashsn0**
Everything is good.

However when running the mainDriver, commands entered using the CommandProcessorDriver actually trigers the command :o 

The CommandProcessorDriver do not perform state transition so the validation should only allows valid commands in start state. However, if used in combination with the rest of the function and using the same GameEngine, we can showcase a working flow of the game.

### CODE REVIEW

File: GameEngine/GameEngine.h [hunk 1]
```diff
@@ -45,7 +46,7 @@ class GameState {
 };
 
 // The GameEngine class manages the game states and processes commands.
-class GameEngine {
+class GameEngine : public Map{
```
Comment (aloisorozco) [inline GameEngine/GameEngine.h:49]: um what? why r u making the gameengine a child of the map class

Comment (MaxGrabs) [reply GameEngine/GameEngine.h:49]: it's supposed to be ILoggable and Subject, idk where that went.

File: Orders/Orders.cpp
Comment (aloisorozco) [inline Orders/Orders.cpp:1]: u reverted all changes here plz fix this pull from main into ur branch

File: Orders/OrdersDriver.cpp
Comment (aloisorozco) [inline Orders/OrdersDriver.cpp:1]: pretty sure u also reverted this

File: GameEngine/GameEngine.h [hunk 2]
```diff
@@ -73,11 +74,21 @@ class GameEngine {
...
     void play();
 
+    void addMap(Map* map);
+
+    Map* getMap() const;
+
     void mainGameLoop();
 
-    void startupPhase();
+    void startupPhase(GameEngine &engineArg);
```
Comment (aloisorozco) [inline GameEngine/GameEngine.h:91]: im so confused why would a game engine method need a game engine reference?

File: GameEngine/GameEngine.h (hunk 1 above)
Comment (aloisorozco) [inline GameEngine/GameEngine.h:49]: this complicates things way too last minute i doubt you tested extensively plz leave how it was before

**Review by aloisorozco — CHANGES_REQUESTED** [review]
check comments i think u changed too many things that all other stuff r dependent on specifically game engine being a child class of map & the gameenging method startupPhase

**Pull request closed without merging**